### PR TITLE
Reintroduce the custom screensaver

### LINKED ISSUE #98: UX: reintroduce our custom screensaver
# Context

Since commit a21e7b721683520e6747f8b2808e52ec17fa6246, the baking application is no longer able to sign baking operations quickly enough.
The signing time for this commit is identical to the previous one (1-2s) but frequently the signing time could increase by 3-6s.

In this commit, our custom `io_event` function, which overrides the one in Ledger's `lib_standard_app`, has been removed.
The difference between these two versions of `io_event` is that `TICKER` events are not handled in our version.
These events have been reactivated in this commit so that Ledger's screensaver can be used.

But it seems that handling `TICKER` events slows down the application.

# Goal

Stop to handle the TICKER event again and reintroduce our customised screensaver.

### PR DESCRIPTION
**Pull request by spalmer25**
fixes #98 
 - Introduce a low cost display
 - Re-introduce the custom screensaver
 - Run the custom screensaver 20s after signing if no user action

### CODE REVIEW

File: src/ui_screensaver.c [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+#include "bolos_target.h"
+#include "globals.h"
+
+#define G_screensaver_state global.dynamic_display.screensaver_state
+
+/**
+ * @brief Black screen element
+ *
+ */
+static const bagl_element_t black_screen_elements[] = {{{BAGL_RECTANGLE,
```
Comment (ajinkyaraj-23) [inline src/ui_screensaver.c:33]: black and blank , why these two variations in naming. Use blank for black_screen_elements too.

**Comment by ajinkyaraj-23** [comment]
Did you check with physical nanos? Whether the screensaver goes on after 20s while signing.

File: src/ui_screensaver.h [hunk 2]
```diff
@@ -0,0 +1,70 @@
+/* Tezos Ledger application - Screen-saver UI functions
+
+   Copyright 2024 TriliTech <contact@trili.tech>
+   Copyright 2024 Functori <contact@functori.com>
+   Copyright 2023 Ledger
```
[the remark this answers is not in the document]
Comment (spalmer25) [reply src/ui_screensaver.h:5]: This code is based on this commit: https://github.com/LedgerHQ/app-tezos/pull/63/commits/3c913cace83b62b4e963c91c04b6b0e52af555a2

**Comment by spalmer25** [comment]
> Did you check with physical nanos? Whether the screensaver goes on after 20s while signing.

Yes, I did. But if you don't mind, it would be safer if you checked it too.

File: src/ui_screensaver.h [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+typedef struct {
+    /// Clock has started
+    bool on;
+    /// Timeout out before saving screen
+    unsigned int timeout;
+} ux_screensaver_clock_t;
+
+typedef struct {
+    /// Screensaver is on/off.
+    bool is_saved;
```
[the remark this answers is not in the document]
Comment (spalmer25) [reply src/ui_screensaver.h]: Done:
[fixup! UX: reintroduce the custom screensaver for NANOS](https://github.com/trilitech/ledger-app-tezos-baking/pull/100/commits/2052fa3227d497ac3f620078128eb1efb6391582)

Rename `is_saved` to `on` in `ux_screensaver_state_t` struct

File: src/ui_screensaver.h [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+
+/**
+ * @brief Empties the screen
+ *
+ *        Waits a click to return to home screen
+ *
+ *        Applies only for Nanos devices
+ *
+ */
+void ui_save_screen(void);
```
[the remark this answers is not in the document]
Comment (spalmer25) [reply src/ui_screensaver.h]: Done:
[fixup! UX: reintroduce the custom screensaver for NANOS](https://github.com/trilitech/ledger-app-tezos-baking/pull/100/commits/947e0aedfa1546835dcb869f1958e1774665331d)

Rename `ui_save_screen` to `ui_start_screensaver`

Will impact: UX: add a screensaver clock

File: src/ui_screensaver.c (hunk 1 above)
Comment (spalmer25) [reply src/ui_screensaver.c]: Done:
[fixup! UX: reintroduce the custom screensaver for NANOS](https://github.com/trilitech/ledger-app-tezos-baking/pull/100/commits/76244fff61193810c9b703299ce3181616f666ed)

Rename `black` to `blank`

**Review by ajinkyaraj-23 — APPROVED** [review]
LGTM

**Pull request merged**